### DISCUSSION TITLE
Display troubles in the Land of AMD Linux

### DISCUSSION THREAD

**Discussion by robertosfield** [vsg-dev/VulkanSceneGraph#1281] [General]
Hi All,

My AMD 5700G Linux system stopped booting at the start of this week, after many attempts to tease it back into life I had to come to the realization that it was a hardware failure, either on the motherboard or the CPU.  So I decided to upgrade motherboard, memory and AMD 8700G CPU to bring myself up to date and get a machine I can test AMD's ray tracing capabilities.

Today I assembled the components, having built quite a few PC's through the years this was pretty straight forward, then I attempted to install Kubuntu 24.04.1 but while it installed the display either went black and stayed off, or flickered on/off with at best 10 seconds of stable operation.  Plenty of others online are reporting these issues but the suggested "fixes" didn't resolve the problems.

I then fell back to installing Kubuntu 22.04 as this had been working really stable for my AMD5700G system but this too has the same issue now.  Whatever has been done with the Kernal/drives/OS configuration has been seriously screwed up for AMD GPUs.  This new machine is unusable as things stand.

Has others in the VSG community come across these issues?  Any suggestions?

**Comment by rolandhill** [comment]
Hi Robert,

I usually have problems with the graphics driver installing Ubuntu. I don't
specifically recall if 24.04 had issues but it is so common that it isn't
notable. Have you tried going into to the advanced menu in grub then
booting in recovery mode? Sometimes just booting recovery can fix the
problem. Otherwise you can install the proprietary divers from there.

Ok, I just tried this to make sure I got the instructions correct and the
screens didn't even switch on in recovery mode. That's never happened
before. Normal boot mode is still working for me. It could be a problem
with the Nouveau driver in my case, with the NVIDIA proprietary driver
working fine.

While this might not be helpful in the end, it indicates that there is a
wider problem.

Cheers,
Roland

**Comment by robertosfield** [comment]
I have had remarkably few issues with drivers in the past few years, being able to use the RADV drivers for my AMD5700G and the AMD CPU I had before it worked pretty well.  My Intel laptop and AMD desktop with integrated GPUs had all been working great. Even when I pop in an NVidia card the NVidia drivers had been reliable too and had got accustomed to Linux just working out of the box on all my hardware. 
 
However, that all came to grinding halt when I installed Kubuntu 24.04 on my AMD5700G system this spring, it run but the display was on/off so much that it was unusable.  There are suggestions on Ubuntu forum's of disabling the splash screen in grub which helped solve the start up blank screen problems but I'd still see issues when using the desktop with the display randomly switching on/off.

The hardware failure earlier this week forcing an upgrade to new hardware and installing an new OS has made it having to resolve this issue inescapable.  Kubuntu 24.04.1 didn't work - same problems I've seen with my AMD5700G had, but on trying out Kubuntu 22.04.5 found exactly the same issue - so it looks like a kernal or driver update to even 22.04 has broken things on AMD graphics.  I am just stunned but how one of the main CPUs uses on Linux machines is so badly broken, it seem like no one is testing stuff on a range of machines before pushing updates out.

Perhaps installing a NVidia graphics card will help.  I'll try that now.

**Comment by rolandhill** [comment]
My screens also turn off randomly for a few seconds. I thought it was just
me. I googled around a bit and didn't find anything helpful. It has only
been happening since I installed Kubuntu 24.04. Mine is a Nvidia card.

**Comment by robertosfield** [comment]
Had hoped that a NVidia card might cure all the problems I'm seeing, but now looks like it's a more widespread regression.  This makes it even more perplexing that it got through testing.

**Comment by robertosfield** [comment]
I now have my new AMD8700G system running with a Geforce 1650 card and Kubuntu 24.04.1, so far the display is working.  I haven't tried install the NVidia drivers yet or installing the dev environment but at least I can see what I'm doing!

Curiously firefox is failing to start due to the configuration issue of snap which I haven't yet got to the bottom of.  Currently writing from my Kubuntu 24.04 laptop - using firefox but not the snap one that comes by default - I had to rip out the snap version with one pulled in from apt-get, but only first by disabling the whole remapping of apt to snap.  Looks like I'll need to look up how to fix that again.

**Comment by robertosfield** [comment]
I now have the VSG building on my new system and initially I didn't install the NVidia drivers and vsgdeviceselection was reporting I was running the RADV driver, which will have been using the AMD5700G's integrated GPU but as the HDMI cable was plugged into my Geforce 1650 card some PCI express must have been happening under the hood!

I was even able to run vsgraytracing example as a test and it worked fine with the RADV driver.  I then installed the NVIdia driver and the system and the VSG are all still working fine, but... vsgraytracing no longer works as the Geforce 1650 doesn't support ray tracing.

The paradoxical thing is the RADV driver running worked really well despite the routing of the display output, no screen dropouts, full resolution and function, but without the Geforce 1650 card, plugging the HMDI cable directly into the motherboards panel the display was so glitchy it was totally unusable.  I know it's software problem as my previous motherbord and AMD5700G worked fine with Kubuntu 22.04 but had the same failure under Kubuntu 24.04 as I'm saw until I plugged in the Geforce 1650.
